application/models/SurveyLanguageSetting.php

```
@@ -359,4 +358,79 @@ public function getAttachmentsData()
+    public function getValidAttachments($exist = true)
+    {
+        return $this->validateAttachments($this->attachments, $exist);
+    }
+
+    /**
+     * Get valid attachements in array
+     * @param string $attachement the attahcment string to be filtered
+     * @param boolean $exist check if file exist in a valid directory
+     * @return string[][][] : array of attachments array by template, sample by key [template][][url,size,relevance]
+     **/
+    public function validateAttachments($string, $exist = true)
```

PHPMD (2.15.0)

[error] The method validateAttachments has a boolean flag argument $exist, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)